### DISCUSSION TITLE
Placeholder Text for the Input Fields

### DISCUSSION THREAD

**Discussion by GeorgeYaz** [espanso/espanso#1164] [Feature requests and Ideas]
**Is your feature request related to a problem? Please describe.**
This feature is not related to any problem, but I guess it could be good for re-usability in case several persons are using your configurations. So normally if we have the following configuration:
```
matches:
  - trigger: "FullName"
    form: "Enter your name here: [[LastName]], [[FirstName]]"
```

This would generate the following:
![image](https://user-images.githubusercontent.com/82532794/173846767-4b97ad6b-b09f-4e08-9be1-4d394c790432.png)
And there you have the white fields empty to be filled by the user, but still the user has no idea where to put the first name and the last name.

**Describe the solution you'd like**
In this matter I was thinking if it would be possible to somehow put a gray default text as a comment, that gives a hint what to enter in these fields, and as soon as you press the field the gray text will disappear and the user can enter his/her text.

For example it would look like this:
![image](https://user-images.githubusercontent.com/82532794/173848080-6d4b3790-578f-493c-992b-a662391a1f76.png)

**Describe alternatives you've considered**
I tried the default property, but apparently the default text is written and you have to remove it manually and of course it's not grayed out because it's a normal text, which could lead to some confusions.

**Comment by dmgolembiowski** [comment]
A potential workaround is to preface the `{` in the yaml file with the friendly-named text. E.g. `First Name: {{first_name}}`

**Reply by GeorgeYaz** [reply]
Hi @dmgolembiowski 

Thank you very much for the suggestion. 
Could you please give an example how to final yaml code would look like?

Thank you!

**Reply by dmgolembiowski** [reply]
> Hi @dmgolembiowski 
> 
> Thank you very much for the suggestion. 
> Could you please give an example how to final yaml code would look like?
> 
> Thank you!

Something like this:

```yaml
- trigger: "FullName" 
  form: "Enter your name here :: Last Name:  [[LastName]], First Name: [[FirstName]]"
```

**Comment by nhasian** [comment]
It would be helpful to have Placeholder Text for the Input Fields as described by @GeorgeYaz .

When making templates for emails or text messages, for example, you cannot preface empty fields with `FirstName` or `UserName` as the recipient will see these.

**Comment by alexsoyes** [comment]
Could be a great feature!

BTW amazing project that I do recommend everywhere ❤️
